src/service.js

```diff
@@ -164,30 +129,35 @@ export function installServiceInEmbedScope(embedWin, id, service) {
   const topWin = getTopWindow(embedWin);
   dev().assert(embedWin != topWin,
       'Service override can only be installed in embed window: %s', id);
-  dev().assert(!getLocalExistingServiceForEmbedWinOrNull(embedWin, id),
+  dev().assert(!getExistingServiceForEmbedWinOrNull(embedWin, id),
       'Service override has already been installed: %s', id);
-  getServiceInternal(embedWin, embedWin, id, undefined, () => service);
+  registerServiceInternal(
+      embedWin,
+      embedWin,
+      id,
+      /* opt_ctor */ undefined,
+      () => service);
+  // Force service to build
+  getServiceInternal(embedWin, embedWin, id);
 }
 
-
 /**
  * @param {!Window} embedWin
  * @param {string} id
  * @return {?Object}
  */
-function getLocalExistingServiceForEmbedWinOrNull(embedWin, id) {
+function getExistingServiceForEmbedWinOrNull(embedWin, id) {
```

Comment:
Why remove the "Local" qualifier? Seems important vis-a-vis `getExistingServiceInEmbedScope` since the former falls back to the top-level window.